File: swagger/models/get_schedule_items_request.py
from .base_model import BaseModel


class GetScheduleItemsRequest(BaseModel):
	"""
	no description available

	Attributes:

		location_ids
		A list of requested location IDs.

		staff_ids
		A list of requested staff IDs.

		start_date
		The start date of the requested date range.   <br />Default:
		**today’s date**

		end_date
		The end date of the requested date range.   <br />Default: **today’s
		date**

		ignore_prep_finish_times
		When `true`, appointment preparation and finish unavailabilities are
		not returned.   <br />Default: **false**

		limit
		Number of results to include, defaults to 100

		offset
		Page offset, defaults to 0.
	"""

	input_map = {
		'LocationIds': 'location_ids',
		'StaffIds': 'staff_ids',
		'StartDate': 'start_date',
		'EndDate': 'end_date',
		'IgnorePrepFinishTimes': 'ignore_prep_finish_times',
		'Limit': 'limit',
		'Offset': 'offset',
		}

	output_map = {
		'location_ids': 'LocationIds',
		'staff_ids': 'StaffIds',
		'start_date': 'StartDate',
		'end_date': 'EndDate',
		'ignore_prep_finish_times': 'IgnorePrepFinishTimes',
		'limit': 'Limit',
		'offset': 'Offset',
		}

File: swagger/Models/GetScheduleItemsRequest.php
<?php

namespace App\MindBody\Models;

/**
 * no description available
 */
class GetScheduleItemsRequest extends BaseModel
{
	/**
	 * A list of requested location IDs.
	 * 
	 * @var integer[] 
	 */
	public $locationIds;
	/**
	 * A list of requested staff IDs.
	 * 
	 * @var integer[] 
	 */
	public $staffIds;
	/**
	 * The start date of the requested date range.   <br />Default:
	 * **today’s date**
	 * 
	 * @var string format:date-time
	 */
	public $startDate;
	/**
	 * The end date of the requested date range.   <br />Default: **today’s
	 * date**
	 * 
	 * @var string format:date-time
	 */
	public $endDate;
	/**
	 * When `true`, appointment preparation and finish unavailabilities are
	 * not returned.   <br />Default: **false**
	 * 
	 * @var boolean 
	 */
	public $ignorePrepFinishTimes;
	/**
	 * Number of results to include, defaults to 100
	 * 
	 * @var integer format:int32
	 */
	public $limit;
	/**
	 * Page offset, defaults to 0.
	 * 
	 * @var integer format:int32
	 */
	public $offset;

	protected function getInputMap()
	{
		return [
			'LocationIds' => ['locationIds', 'array', null],
			'StaffIds' => ['staffIds', 'array', null],
			'StartDate' => ['startDate', 'string', 'date-time'],
			'EndDate' => ['endDate', 'string', 'date-time'],
			'IgnorePrepFinishTimes' => ['ignorePrepFinishTimes', 'boolean', null],
			'Limit' => ['limit', 'integer', 'int32'],
			'Offset' => ['offset', 'integer', 'int32'],
			];
	}

	protected function getOutputMap()
	{
		return [
			'locationIds' => ['LocationIds', 'array', null],
			'staffIds' => ['StaffIds', 'array', null],
			'startDate' => ['StartDate', 'string', 'date-time'],
			'endDate' => ['EndDate', 'string', 'date-time'],
			'ignorePrepFinishTimes' => ['IgnorePrepFinishTimes', 'boolean', null],
			'limit' => ['Limit', 'integer', 'int32'],
			'offset' => ['Offset', 'integer', 'int32'],
			];
	}
}
